enumerable_methods.rb

```diff
@@ -0,0 +1,126 @@
+module Enumerable
+  def my_each
+    return enum_for(:my_each) unless block_given?
+
+    i = 0
+    while i < size
+      yield(self[i])
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    return enum_for(:my_each_with_index) unless block_given?
+
+    i = 0
+    my_each do |item|
+      yield(item, i)
+      i += 1
+    end
+  end
+
+  def my_select
+    return enum_for(:my_select) unless block_given?
+
+    result = []
+    i = 0
+    while i < size
+      result << self[i] if yield(self[i]) == true
+      i += 1
+    end
+  end
+
+  def my_all?(args = nil)
+    condition = true
+    if block_given?
+      my_each { |item| condition = false if yield(item) == false }
+    elsif args.is_a?(Class)
+      my_each { |item| condition = false if item.is_a?(args) }
+    elsif args.is_a?(Regexp)
+      my_each { |item| condition = false if args.match?(item.to_s) }
+    elsif !args.nil?
+      my_each { |item| condition = false if item != args }
+    else
+      my_each { |item| condition = false if item }
+    end
+    condition
+  end
```

Comment:
my_all?

- when no block or argument is given returns true when none of the collection members are false or nil

`[nil, true, 99].all?` return false
`[2, 5, 'a'].all?` return true   ===>  **your code returns false here**



- when a class is passed as an argument returns true if all of the collection is a member of such class

`[1, 2i, 3.14].all?(Numeric)` returns true



- when a Regex is passed as an argument returns true if all of the collection matches the Regex

`%w[ant bear cat].all?(/t/)`  returns false
`%w[ant bit boot].all?(/t/)` returns true